### PR TITLE
Update SmallRye GraphQL to 2.8.0

### PR DESCRIPTION
**Pull request by mskacelik**
/cc @jmartisk

### CODE REVIEW

**Review by jmartisk — APPROVED** [review]

File: extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLConfig.java [hunk 1]
```diff
@@ -221,4 +221,12 @@ public class SmallRyeGraphQLConfig {
      */
     @ConfigItem
     public Optional<List<ExtraScalar>> extraScalars;
+
+    /**
+     * Excludes all the 'null' fields in the GraphQL response's <code>data</code> field,
+     * except for the non-successfully resolved fields (errors).
+     * Disabled by default.
+     */
+    @ConfigItem(name = "exclude-null-fields-in-responses")
```
Comment (gsmet) [inline extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLConfig.java:230]: Note that you don't need to define a `name` in the case where the convention is OK, which seems the case here?

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/GraphQLClientBuildConfig.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+@ConfigRoot(name = "smallrye-graphql-client", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class GraphQLClientBuildConfig {
+
+    /**
+     * Configuration item to enable build-time scanning in Quarkus for generating typesafe GraphQL client models.
+     * If true, build-time scanning is enabled. By default, it is true.
+     */
+    @ConfigItem(name = "enable-build-time-scanning", defaultValue = "true")
+    public boolean allowBuildTimeScanning;
```
Comment (jmartisk) [inline extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/GraphQLClientBuildConfig.java:15]: Hmm, maybe rename  the field to `enableBuildTimeScanning` and remove the explicit name above.. 'Allow' sounds like it has something to do with security, which it doesn't..

Comment (gsmet) [reply extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/GraphQLClientBuildConfig.java]: Yeah in general, let's try to be consistent and avoid the `name` attribute when you can.

**Review by jmartisk — APPROVED** [review]

**Comment by jmartisk** [comment]
@gsmet looks good now?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Ah yes, sorry I didn't want to make it a binding review, just wanted to add a comment review.

Let's merge it.

**Pull request merged**